Lib/_strptime.py

```diff
@@ -371,7 +371,10 @@ def __init__(self, locale_time=None):
             # W is set below by using 'U'
             'y': r"(?P<y>\d\d)",
             'Y': r"(?P<Y>\d\d\d\d)",
+            # See gh-121237: "z" might not support colons, if designed from scratch.
+            # However, for backwards compatibility, we must keep them.
```

Comment:
```suggestion
            # See gh-121237: "z" must support colons for backwards compatibility.
```

"'z' might not support colons" could be interpreted as meaning that it is possible that `z` doesn't support colons, but in this context it it supposed to mean "We might not have chosen to support colons...". I suggest going with a simpler wording.